extensions/jni/jvm/JniReferenceObjects.h

```diff
@@ -128,8 +128,8 @@ class JniByteInputStream {
     return read;
   }
 
-  int64_t read(uint8_t &arr) {
-    return stream_->read(arr);
+  int64_t read(uint8_t &arr) const {
+    return gsl::narrow<int64_t>(stream_->read(arr));
```

Comment:
If `stream_` is a `minifi::io::InputStream`, then `read()` can return `static_cast<size_t>(-1)`, which will cause this to terminate (or return an incorrect value if `size_t` is smaller than 64 bits).  We need to check if the return value `io::isError()`, and return `-1` explicitly in that case.

Reply:
Nice catch, I've added the error check in https://github.com/apache/nifi-minifi-cpp/pull/1561/commits/dd896082c79acc2b122a3e74ab36c527f1fee446